src/Compilers/CSharp/Portable/Lowering/IteratorRewriter/IteratorRewriter.cs

```diff
@@ -74,7 +75,91 @@ internal partial class IteratorRewriter : StateMachineRewriter
 
             stateMachineType = new IteratorStateMachine(slotAllocatorOpt, compilationState, method, methodOrdinal, isEnumerable, elementType);
             compilationState.ModuleBuilderOpt.CompilationState.SetStateMachineType(method, stateMachineType);
-            return new IteratorRewriter(body, method, isEnumerable, stateMachineType, slotAllocatorOpt, compilationState, diagnostics).Rewrite();
+            var rewriter = new IteratorRewriter(body, method, isEnumerable, stateMachineType, slotAllocatorOpt, compilationState, diagnostics);
+            if (rewriter.EnsureAllSymbolsAndSignature())
+            {
+                return body;
+            }
+
+            return rewriter.Rewrite();
+        }
+
+        // returns true if any type we depend on is not present
+        protected bool EnsureAllSymbolsAndSignature()
```

Comment:
Consider renaming to "VerifyPresenceOfRequiredAPIs" and changing to return true in case of success.